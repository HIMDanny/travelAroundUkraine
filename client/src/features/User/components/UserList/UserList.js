import React from 'react';
import { Typography, Paper, Box } from '@mui/material';
import PropTypes from 'prop-types';

const ListSX = {
  display: 'flex',
  flexDirection: 'column',
  borderRadius: '20px',
  maxWidth: '400px',
  minWidth: '250px',
  py: '20px',
  px: 2,
  listStyleType: 'none',
  m: '2% auto',
};

const listItemSX = {
  m: { xs: '15px' },
};

const listItemTitleSX = {
  color: 'primary.main',
  fontSize: '20px',
};

const listItemInnerTextSX = { my: '2px', fontSize: '18px', mx: { tablet: '20px' } };

const UserList = ({ userData: { firstName, email, telephone, gender } }) => (
  <Paper component="ul" sx={ListSX}>
    <Typography sx={{ m: '15px auto', fontSize: '24px' }} component="h3">
      {`${firstName}'s info`}
    </Typography>
    <Box sx={listItemSX} component="li">
      <Typography sx={listItemTitleSX}>Email: </Typography>
      <Typography sx={listItemInnerTextSX}>{email} </Typography>
    </Box>
    <Box sx={listItemSX} component="li">
      <Typography sx={listItemTitleSX}>Phone: </Typography>
      <Typography sx={listItemInnerTextSX}>{telephone} </Typography>
    </Box>
    <Box sx={listItemSX} component="li">
      <Typography sx={listItemTitleSX}>Gender: </Typography>
      <Typography sx={listItemInnerTextSX}>{gender} </Typography>
    </Box>
  </Paper>
);

UserList.propTypes = {
  userData: PropTypes.object.isRequired,
};

export default UserList;
