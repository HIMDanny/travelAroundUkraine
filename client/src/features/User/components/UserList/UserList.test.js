import { render } from '@testing-library/react';
import UserList from './UserList';

jest.mock('@mui/material/Typography', () => ({ children }) => <li>{children}</li>);
jest.mock('@mui/material/Box', () => ({ children }) => <ul>{children}</ul>);
jest.mock('@mui/material/Paper', () => ({ children }) => <div>{children}</div>);

const userData = {
  firstName: 'Roman',
  email: 'Rymarchuk@gmail.com',
  telephone: '38000000000',
  gender: 'male',
};

describe('Render UserList', () => {
  test('should UserList render', () => {
    const { asFragment } = render(<UserList userData={userData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
