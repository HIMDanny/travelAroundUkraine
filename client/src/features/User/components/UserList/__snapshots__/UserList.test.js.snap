// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserList should UserList render 1`] = `
<DocumentFragment>
  <div>
    <li>
      Roman's info
    </li>
    <ul>
      <li>
        Email: 
      </li>
      <li>
        Rymarchuk@gmail.com 
      </li>
    </ul>
    <ul>
      <li>
        Phone: 
      </li>
      <li>
        38000000000 
      </li>
    </ul>
    <ul>
      <li>
        Gender: 
      </li>
      <li>
        male 
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
